src/Helper/DefaultImplementationLocator.php

```diff
@@ -68,40 +52,35 @@ final class DefaultImplementationLocator
 
     public function findHttpClient(): ?ClientInterface
     {
-        foreach ($this->defaultImplementations['client'] as $class => $config) {
-            if (class_exists($class)) {
-                $result = $this->constructObject($class, $config);
-                if (!$result instanceof ClientInterface) {
-                    // @codeCoverageIgnoreStart
-                    throw new LogicException('The resulting object is not an instance of ' . ClientInterface::class);
-                    // @codeCoverageIgnoreEnd
-                }
-
-                return $result;
-            }
+        try {
+            /**
+             * Discovery triggers an error if symfony/http-client is installed
+             * and php-http/httplug is not, even if the intention is to find a
+             * PSR-18 client. Since the discovery will otherwise be successful,
+             * let's silence the error.
+             */
+            return @Psr18ClientDiscovery::find();
+        } catch (NotFoundException $e) {
+            return null;
         }
-
-        return null;
     }
 
     public function findRequestFactory(): ?RequestFactoryInterface
     {
-        foreach ($this->defaultImplementations['factory'] as $class => $config) {
-            if (class_exists($class)) {
-                $result = $this->constructObject($class, $config);
-                if (!$result instanceof RequestFactoryInterface) {
-                    // @codeCoverageIgnoreStart
-                    throw new LogicException(
-                        'The resulting object is not an instance of ' . RequestFactoryInterface::class
-                    );
-                    // @codeCoverageIgnoreEnd
-                }
-
-                return $result;
-            }
+        try {
+            return Psr17FactoryDiscovery::findRequestFactory();
+        } catch (NotFoundException $e) {
+            /**
+             * This will only be thrown if a HTTP client was found, but a request factory is not.
+             * Due to how php-http/discovery works, this scenario is unlikely to happen.
+             * See linked comment for more info.
+             *
+             * https://github.com/Unleash/unleash-client-php/pull/27#issuecomment-920764416
+             */
+            // @codeCoverageIgnoreStart
```

Comment:
The exception catching should be marked as uncovered as well, otherwise it's reported as missing coverage.

```suggestion
            return Psr17FactoryDiscovery::findRequestFactory();
            // @codeCoverageIgnoreStart
        } catch (NotFoundException $e) {
            /**
             * This will only be thrown if a HTTP client was found, but a request factory is not.
             * Due to how php-http/discovery works, this scenario is unlikely to happen.
             * See linked comment for more info.
             *
             * https://github.com/Unleash/unleash-client-php/pull/27#issuecomment-920764416
             */
```